api/v3/Activity.php

```diff
@@ -470,10 +470,16 @@ function _civicrm_api3_activity_get_formatResult($params, $activities, $options)
         break;
 
       case 'case_id':
-        $dao = CRM_Core_DAO::executeQuery("SELECT activity_id, case_id FROM civicrm_case_activity WHERE activity_id IN (%1)",
+      case 'case_subject':
+        $dao = CRM_Core_DAO::executeQuery("SELECT activity_id, case_id, subject FROM civicrm_case_activity cca INNER JOIN civicrm_case cc ON cc.id = cca.case_id WHERE activity_id IN (%1)",
```

Comment:
I think you are right, lemme undo my changes for this and use ```CRM_Core_DAO::executeQuery('CRM_Case_DAO_Case', $activities['case_id'], 'subject')``` instead, to retrieve the case subject for this.